graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java

```diff
@@ -200,6 +237,28 @@ public ConfigurationRequest getRequestedConfiguration() {
                             TextField.Attribute.IS_PASSWORD
                     )
             );
+            x.addField(
+                    new BooleanField(
+                            CK_TLS_NEED_CLIENT_AUTH,
+                            "TLS Need Client Auth",
+                            false,
+                            "TLS Need Client Auth")
```

Comment:
Okay, the confusion is solved.

The "want" option accepts clients without any cert and clients with a valid cert. Clients with invalid certs are not accepted.

The "need" option only accepts clients with a valid certificate.

The code needs to be changed to only set one of the options because they overwrite each other. @joschi is currently working on that.